src/core/sock.c

```diff
@@ -122,6 +122,7 @@ int nn_sock_init (struct nn_sock *self, struct nn_socktype *socktype, int fd)
     self->linger = 1000;
     self->sndbuf = 128 * 1024;
     self->rcvbuf = 128 * 1024;
+    self->rcvmaxsize = 512 * 1024 * 1024;
```

Comment:
I'd stick with 1MB.   Even 10MB is too big.   In some circumstances 1MB is too big.  (Imagine opening up a 1000 sockets and sending one of these to each one.  Congratulations -- you've allocated 1GB on the server.)

For internet facing frames, I'd choose a much smaller value < 100K I think, and also limit the number of connections overall.  We lack the ability to limit.  (The limit could be imposed per peer, but that means quite a bit more work.)